standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java

```diff
@@ -3959,7 +3959,7 @@ public List<Partition> listPartitionsPsWithAuth(String catName, String db_name,
         LOG.info(
             "Redirecting to directSQL enabled API: db: {} tbl: {} partVals: {}",
             db_name, tbl_name, Joiner.on(',').join(part_vals));
-        return getPartitions(catName, db_name, tbl_name, -1);
+        return getPartitions(catName, db_name, tbl_name, max_parts);
```

Comment:
Is this change related?

Reply:
Will fix it in a separate PR: https://github.com/apache/hive/pull/4662

Reply:
Fixed.